app/views/collaborators/_collaborator.html.erb

```diff
@@ -4,14 +4,13 @@
     <%# this should more properly use a variable set in your controller rather than User.all - for example you might want to limit the list of possible collaborators according to some condition %>
     <% User.all.each do |user| %>
       <tr>
-        <% if user != wiki.user %>
+        <% if wiki.private? && user != wiki.user && current_user == wiki.user && !(user.collaborator_for(wiki)) %>
           <td><%= user.name %></td>
+          <td><%= link_to  "make_collaborator", collaborators_path(wiki_id: wiki.id, user_id:user.id), class: 'btn btn-primary', method: :post %></td>
         <% end %>
-        <% if wiki.private? && user != wiki.user && !(collaborator = user.collaborator_for(wiki, user)) %>
-          <td><%= link_to  "make collaborator", collaborators_path(wiki_id: wiki.id, user_id:user.id), class: 'btn btn-primary', method: :post %></td>
-        <% end %>
-        <% if wiki.private? && user != wiki.user && current_user == wiki.user && (collaborator = user.collaborator_for(wiki,user)) %>
-          <td><%= link_to "Remove_collaborator", collaborator_path(user.collaborator_for(wiki,user)), class: 'btn btn-danger', method: :delete %></td>
+        <% if wiki.private? && user != wiki.user && current_user == wiki.user && (user.collaborator_for(wiki)) %>
+          <td><%= user.name %></td>
```

Comment:
See feedback on earlier PR and let me know if there's anything needing clarification: https://github.com/sharadalt/blocipedea/pull/14#discussion_r58203663